frontend/src/app/modules/interventions/list/table/table.component.ts

```diff
@@ -1,44 +1,63 @@
-import { Component, Input, ViewChild, OnChanges, AfterViewInit } from '@angular/core';
+import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
 import { Router } from '@angular/router';
-import { MatTable, MatTableDataSource, MatPaginator } from '@angular/material';
+import { MatPaginator, MatSort, MatTable } from '@angular/material';
 
-import { Intervention } from '@shared/domain/intervention.model';
+import { Intervention, SortDirection, InterventionsFilter } from '@shared/domain/intervention.model';
+import { InterventionsDatasource } from './interventions.datasource';
+import { InterventionsService } from '../../interventions.service';
+import { tap } from 'rxjs/operators';
+import { merge } from 'rxjs';
+
+const defaultSearchParams: InterventionsFilter = {
+  pageSize: 10,
+  page: 1,
+  sortBy: 'creationDate',
+  sortDirection: SortDirection.Descending,
+};
 
 @Component({
   selector: 'app-interventions-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss'],
 })
-export class TableComponent implements OnChanges, AfterViewInit {
+export class TableComponent implements AfterViewInit, OnInit {
```

Comment:
You are adding some subscriptions which should be unsubscribed on destroy, as they will cause a memory leak (on of the leaks here would be very severe) if not cleaned up properly.

It is not much to write, but you don't need to write it by hand, just extend 'ComponentWithSubscriptions' utility class that I wrote last summer, and then use `this.subscriptions.add(...subscription goes here...)`

Reply:
OK, done 👍